ISSUE 1381: Prometheus metrics were not exported correctly

IdoZilberberg opened the pull request:
Fixes #1381

netoneko commented on instrumentation/metric/histogram_prometheus_test.go:
```diff
@@ -8,36 +8,59 @@
 
-	resultWithParams := r.ExportPrometheus()
-	require.Regexp(t, "Some_Latency{vcid=\"100000\",quantile=\"0.01\"} 0.00", resultWithParams)
+	//status := r.NewHistogram("Some.Latency", 10000)
```
Please remove commented out code

IdoZilberberg replied:
Removed.

electricmonk commented on instrumentation/metric/prometheus.go:
```diff
@@ -26,25 +26,29 @@ func prometheusName(name string) string {
 
-func (r *prometheusRow) String(pairs ...prometheusKeyValuePair) string {
-	return r.name + r.wrapParams(pairs...) + " " + r.value
+func QuantileAsStr(quantile float64) string {
```
why is this exported?

IdoZilberberg replied:
Fixed

IdoZilberberg commented on instrumentation/metric/histogram_prometheus_test.go:
```diff
@@ -8,36 +8,59 @@
+	fmt.Println(prometheusStrings)
+	require.Regexp(t, "# TYPE Some_Latency histogram", promStr)
+	for _, row := range metrics["Some.Latency"].PrometheusRow() {
+		expectedStr := fmt.Sprintf("Some_Latency{quantile=\"%s\"} %s",
```
I like this better as regexp, it is more resilient. If I were to compare an exact string match, it might break on newlines, spaces and such.

IdoZilberberg commented:
@electricmonk @netoneko There were 2 histogram tests - one that has no `vcid` label and another that does. I felt this is redundant so I kept only the one that has `vcid`.
LMK if you think otherwise. I'll wait before merging it.

IdoZilberberg commented:
@netoneko I renamed `quantile` to `aggregation` and added missing aggregations.
See: https://github.com/orbs-network/orbs-network-go/issues/1381#issuecomment-545402749

^ @electricmonk
